org.envirocar.app/src/org/envirocar/app/views/dashboard/DashboardFragment.java

```diff
@@ -444,16 +447,31 @@ protected void onStartTrackButtonClicked() {
                         BluetoothDevice device = bluetoothHandler.getSelectedBluetoothDevice();
 
                         Intent obdRecordingIntent = new Intent(getActivity(), RecordingService.class);
-                        this.connectingDialog = new MaterialDialog.Builder(getActivity())
-                                .iconRes(R.drawable.ic_bluetooth_searching_black_24dp)
-                                .title(R.string.dashboard_connecting)
-                                .content(String.format(getString(R.string.dashboard_connecting_find_template), device.getName()))
-                                .progress(true, 0)
-                                .negativeText(R.string.cancel)
-                                .cancelable(false)
-                                .onNegative((dialog, which) -> getActivity().stopService(obdRecordingIntent))
+
+                        this.connectingDialog = DialogUtils.createDefaultDialogBuilder(getContext(),
+                                R.string.dashboard_connecting,
+                                R.drawable.ic_bluetooth_white_24dp,
+                                String.format(getString(R.string.dashboard_connecting_find_template), device.getName()))
+                                .setNegativeButton(R.string.cancel,(dialog, which) -> {
+                                    getActivity().stopService(obdRecordingIntent);
+                                })
                                 .show();
 
+                        // If the device is not found to start the track, dismiss the Dialog in 15 sec
+                        new CountDownTimer(15000, 1000) {
+                            @Override
+                            public void onTick(long millisUntilFinished) {
+                            }
+                            @Override
+                            public void onFinish() {
+                                connectingDialog.dismiss();
+                                getActivity().stopService(obdRecordingIntent);
```

Comment:
You could also use the existing `ServiceUtils` class with `  ServiceUtils.stopService(...)`.

Reply:
Initially, i also used `ServiceUtils.stopService()` to stop the recording but got some error's in `ServiceUtils `class. 

![Screenshot from 2021-06-23 15-50-57](https://user-images.githubusercontent.com/70392921/123080767-e596f780-d43a-11eb-8328-473b7b858d45.png)

Reply:
Have a look on how it's done with `startService()` in the `ServiceUtils` class. Maybe, it's helpful to overload `StopService` in a similar manner.